### PR TITLE
[IRRC-83] Create modal to show suggested teachers 

### PR DESCRIPTION
**Pull request by dzldebby**

### CODE REVIEW

File: backend/src/routes/matching.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      } else {
+        return res.status(500).send({
+          message: err.message,
+          type: 'UnknownError'
+        })
+      }
+    }
+    return res.status(200).json(result)
+  }
+  return res.status(500).send({
```
Comment (thchia) [inline backend/src/routes/matching.js]: You can use `422` here. It's more specific.

File: frontend/src/pages/Matching.vue [hunk 2]
```diff
@@ -219,23 +222,23 @@
...
                 >
-                  <span>Unmatch Selected </span> ({{ checkedRows.length }})
+                  <span>Unmatch Selected </span> ({{ selectedMatches.length }})
                 </button>
                 <div class="divider"/>
                 <button
                   class="button field is-white"
-                  @click="cardModal()"
-                  :disabled="!checkedRows.length"
+                  @click="confirmSelected()"
```
Comment (thchia) [inline frontend/src/pages/Matching.vue:233]: If we use the same function `confirmSelected` for both "Confirm Selected" and "Confirm All", we will need a way to differentiate when each button is clicked. Consider:

- "Confirm All" means confirm all the rows in `matchedPairs`, regardless of whether they are selected or not;
- "Confirm Selected" means confirm the rows in `matchedPairs` which are also selected.

So we need a way to tell the `confirmSelected` function whether or not to care about the selected status.

**Review by thchia — APPROVED** [review]
Some minor comments but nothing major to block this.

File: frontend/src/pages/Matching.vue [hunk 3]
```diff
@@ -245,55 +257,96 @@
...
-      matchingData: [
-        {'studentName':'Kaasm Rujuthan', 'teacherName':'Praveen Kumar'},
-        {'studentName':'Rujuthan Kaasm', 'teacherName':'Kumar Praveen'},
-      ],
-      selected: null,
+      isComponentModalActive: false,
+      teacherData: [
+        {'teacherName':'Praveen Kumar', 'nativeLanguage':'Bangla', 'secondLanguage':'Tamil', 'company': 'SMU'},
+        {'teacherName':'Kumar Praveen', 'nativeLanguage':'Hindi', 'secondLanguage':'Bangla', 'company': 'SMU'},
+      ],
```
Comment (thchia) [inline frontend/src/pages/Matching.vue]: This should come from the Vuex store.

**Review by thchia — CHANGES_REQUESTED** [review]
This looks like still work in progress (quite a few things changed whilst I was doing the first review)? If so, you can mark it as draft first.

File: frontend/src/pages/Matching.vue (hunk 3 above)
Comment (therealyingtong) [reply frontend/src/pages/Matching.vue]: We were missing a few commits after a rebase. That's been fixed now :+1:

File: backend/src/routes/matching.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+const router = express.Router()
+// const debug = require('debug')('app:students')
+const statuses = require('../helpers/statuses.js')
+const statusUpdates = require('../helpers/statusUpdates.js')
+const { NotFoundError } = require('objection')
+
+// TODO: Handle errors
+
+/* POST multiple students to screening endpoint */
+// on Postman: http://localhost:3001/students/1
```
Comment (therealyingtong) [inline backend/src/routes/matching.js]:
```suggestion
/* POST multiple students to matching endpoint */
// on Postman: http://localhost:3001/matching
```

File: backend/src/routes/matching.js (hunk 1 above)
Comment (therealyingtong) [reply backend/src/routes/matching.js]: Done.

File: frontend/src/pages/Matching.vue (hunk 2 above)
Comment (therealyingtong) [reply frontend/src/pages/Matching.vue:233]: Added `confirmAll()` method.

File: frontend/src/pages/Matching.vue [hunk 5]
```diff
@@ -354,28 +393,69 @@ export default {
...
     },
+    confirmAll() {
+      this.$buefy.dialog.confirm({
+        type: 'is-blue',
+        message: 'All matches will be confirmed. All teachers matched will receive an email in their inbox',
+      })
+    }
   },
+  mounted() {
+    this.getAllTeachers()
```
Comment (thchia) [inline frontend/src/pages/Matching.vue]: I suggest we put this in App.vue like we did for getAllStudents.

**Review by thchia — APPROVED** [review]

**Review by xuanjane — APPROVED** [review]

**Pull request merged**